src/main/kotlin/com/openlattice/mechanic/upgrades/AdjustNCRICDataDateTimes.kt

```diff
@@ -0,0 +1,78 @@
+package com.openlattice.mechanic.upgrades
+
+import com.openlattice.mechanic.Toolbox
+import com.openlattice.postgres.PostgresArrays
+import com.openlattice.postgres.PostgresColumn.*
+import com.openlattice.postgres.PostgresTable.DATA
+import org.slf4j.LoggerFactory
+
+const val DATETIME_COL = "n_timestamptz"
+
+class AdjustNCRICDataDateTimes(private val toolbox: Toolbox) : Upgrade {
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(AdjustNCRICDataDateTimes::class.java)
+    }
+
+    override fun getSupportedVersion(): Long {
+        return Version.V2019_12_12.value
+    }
+
+    override fun upgrade(): Boolean {
+
+        val entitySetsToPropertyTypes = mapOf(
+                "NCRICNotifications" to "general.datetime",
+                "NCRICResultsIn" to "general.datetime",
+                "NCRICVehicleRecords" to "ol.datelogged",
+                "NCRICRecordedBy" to "ol.datelogged",
+                "NCRICIncludes" to "date.completeddatetime"
+        )
+
+        val rangeClausesToOffsets = mapOf(
+                " AND $DATETIME_COL >= '2019-11-03 10:00:00.000000-00' " to 8, // PST
```

Comment:
If the goal here is just to convert some timestamps that were inserted with UTC instead of some other time zone, you can do that with a postgresql expression:

`column at time zone 'UTC' at time zone 'America/Los_Angeles';`

Reply:
```
postgres=# select timestamp with time zone '2019-11-03 10:00:00.000000-00' at time zone 'UTC' at time zone 'America/Los_Angeles';
        timezone
------------------------
 2019-11-03 10:00:00-08
```

Reply:
nope -- the dates were all integrated incorrectly, so now we need to actually modify all the data. buuut turns out citus makes the fix a bit more complicated than this, so further changes are coming...

http://docs.citusdata.com/en/v9.0/reference/common_errors.html#stable-functions-used-in-update-queries-cannot-be-called-with-column-references